build/wasi.cmake

```diff
@@ -0,0 +1,207 @@
+project (
+    uvwasi
+    DESCRIPTION "WASI syscall API built atop libuv"
+    VERSION 0.0.19
+    LANGUAGES C
+)
+
+# Point CMake at any custom modules we may ship
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
+
+# This can be a commit hash or tag
+set(LIBUV_VERSION v1.44.2)
+
+include(CMakeDependentOption)
+cmake_dependent_option(UVWASI_BUILD_TESTS
+  "Build the unit tests when uvwasi is the root project" ON
+  "CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR" OFF)
+
+if(CMAKE_C_COMPILER_ID MATCHES "AppleClang|Clang|GNU")
+  list(APPEND uvwasi_cflags -fvisibility=hidden --std=gnu89)
+  list(APPEND uvwasi_cflags -Wall -Wsign-compare -Wextra -Wstrict-prototypes)
+  list(APPEND uvwasi_cflags -Wno-unused-parameter)
+endif()
+
+set( CMAKE_C_FLAGS -fPIC)
+
+if(DEFINED WALRUS_ARCH)
+  if(${WALRUS_ARCH} STREQUAL "x86")
+    set( CMAKE_C_FLAGS "-fPIC -m32")
+  endif()
+endif()
+
+if(DEFINED WALRUS_MODE)
+  if(${WALRUS_MODE} STREQUAL "debug")
+    set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g") 
+  endif()
+endif()
+
+if(APPLE)
+   set(CMAKE_MACOSX_RPATH ON)
+endif()
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  list(APPEND uvwasi_defines _GNU_SOURCE _POSIX_C_SOURCE=200112)
+endif()
+
+include(FetchContent)
+## https://libuv.org
+FetchContent_Declare(
+        libuv
+        GIT_REPOSITORY https://github.com/libuv/libuv.git
+        GIT_TAG ${LIBUV_VERSION})
+FetchContent_GetProperties(libuv)
+if(NOT libuv_POPULATED)
+    FetchContent_Populate(libuv)
+    include_directories("${libuv_SOURCE_DIR}/include")
+    add_subdirectory(${libuv_SOURCE_DIR} ${libuv_BINARY_DIR} EXCLUDE_FROM_ALL)
+endif()
+set(LIBUV_INCLUDE_DIR ${libuv_SOURCE_DIR}/include)
+set(LIBUV_LIBRARIES uv_a)
+
+## uvwasi source code files.
+set(uvwasi_sources
+    third_party/uvwasi/src/clocks.c
+    third_party/uvwasi/src/fd_table.c
+    third_party/uvwasi/src/path_resolver.c
+    third_party/uvwasi/src/poll_oneoff.c
+    third_party/uvwasi/src/sync_helpers.c
+    third_party/uvwasi/src/uv_mapping.c
+    third_party/uvwasi/src/uvwasi.c
+    third_party/uvwasi/src/wasi_rights.c
+    third_party/uvwasi/src/wasi_serdes.c
+)
+
+option(UVWASI_DEBUG_LOG "Enable debug logging" OFF)
+if(UVWASI_DEBUG_LOG)
+    list(APPEND uvwasi_cflags -DUVWASI_DEBUG_LOG)
+endif()
+
+# Code Coverage Configuration
+add_library(coverage_config INTERFACE)
+
+option(CODE_COVERAGE "Enable coverage reporting" OFF)
+if(CODE_COVERAGE AND CMAKE_C_COMPILER_ID MATCHES "AppleClang|GNU|Clang")
```

Comment:
Would you simplify this `wasi.cmake` file?
For example, we currently don't need any test or analysis about wasi, so we can remove coverage, asan and test build etc.

Reply:
Other looks fine to me :)

Reply:
I have simplified `wasi.cmake` significantly, but left in the debug logging option, beacuse it may come usefull during development, though set it OFF initially. Anything other than that that is not crucial is removed.